archive/multisource.go

```diff
@@ -185,25 +192,58 @@ func (cms *chunkMultiSource) SourceFromRequest(context.Context, *api.WorkerReque
 	return nil, errReqForChunk
 }
 
+func (m *chunkMultiSource) Stats() ChunkStats {
+	return m.stats.Copy()
+}
+
 type chunkSource struct {
 	cms   *chunkMultiSource
 	chunk Chunk
+	stats *ChunkStats
 }
 
 func (s *chunkSource) Open(ctx context.Context, zctx *resolver.Context, sf driver.SourceFilter) (driver.ScannerCloser, error) {
+	var size int64
 	paths := make([]string, len(s.cms.altPaths))
 	for i, input := range s.cms.altPaths {
-		paths[i] = s.chunk.Localize(s.cms.ark, input).String()
+		u := s.chunk.Localize(s.cms.ark, input)
+		stat, err := iosrc.Stat(ctx, u)
+		if err != nil {
+			return nil, err
+		}
+		size += stat.Size()
+		paths[i] = u.String()
 	}
 	rc := detector.MultiFileReader(zctx, paths, zio.ReaderOpts{Format: "zng"})
 	sn, err := scanner.NewScanner(ctx, rc, sf.Filter, sf.FilterExpr, sf.Span)
 	if err != nil {
 		rc.Close()
 		return nil, err
 	}
+	s.stats.Append(ChunkStats{size, size})
 	return &scannerCloser{Scanner: sn, Closer: rc}, nil
 }
 
 func (s *chunkSource) ToRequest(*api.WorkerRequest) error {
 	return errReqForChunk
 }
+
+type ChunkStats struct {
+	// ChunksOpenedSize is the cumulative size of all the chunks read.
+	ChunksOpenedSize int64
+	// ChunksReadSize is the amount of bytes read from all chunks. If seek
+	// indicies are used this number should be less than OpenedChunkSize.
+	ChunksReadSize int64
+}
+
+func (s *ChunkStats) Append(a ChunkStats) {
```

Comment:
Name this like scanner.ScannerStats.Accumulate.
```suggestion
func (s *ChunkStats) Accumulate(a ChunkStats) {
```

Reply:
Better